test/server/admin/clusters_params_test.cc

```diff
@@ -0,0 +1,64 @@
+#include "source/common/buffer/buffer_impl.h"
+#include "source/server/admin/clusters_params.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Server {
+namespace {
+
+TEST(ClustersParamsTest, FormatDefaultsToTextWhenNotSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters";
+  ClustersParams::Format expected_format = ClustersParams::Format::Text;
+  Http::Code expected_code = Http::Code::OK;
+
+  Http::Code code = params.parse(url, buffer);
+
+  EXPECT_EQ(params.format_, expected_format);
+  EXPECT_EQ(code, expected_code);
+}
+
+TEST(ClustersParamsTest, FormatSetToTextWhenTextSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters?format=text";
+  ClustersParams::Format expected_format = ClustersParams::Format::Text;
+  Http::Code expected_code = Http::Code::OK;
+
+  Http::Code code = params.parse(url, buffer);
+
+  EXPECT_EQ(params.format_, expected_format);
+  EXPECT_EQ(code, expected_code);
+}
+
+TEST(ClustersParamsTest, FormatSetToJsonWhenJsonSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters?format=json";
+  ClustersParams::Format expected_format = ClustersParams::Format::Json;
+  Http::Code expected_code = Http::Code::OK;
+
+  Http::Code code = params.parse(url, buffer);
+
+  EXPECT_EQ(params.format_, expected_format);
+  EXPECT_EQ(code, expected_code);
+}
+
+TEST(ClustersParamsTest, FormatSetToUnknownWhenInvalidFormatSupplied) {
+  ClustersParams params;
+  Buffer::OwnedImpl buffer;
+  std::string url = "localhost:1337/clusters?format=fail";
+  ClustersParams::Format expected_format = ClustersParams::Format::Unknown;
```

Comment:
I do see you have a test for Unknown, but Unknown is only a value that we expect to read in tests, never in production, as the status-code check will block reading of format_ in that case.